source/core_http_client.c

```diff
@@ -1723,19 +1732,9 @@ static HTTPStatus_t sendHttpHeaders( const TransportInterface_t * pTransport,
 
     if( returnStatus == HTTPSuccess )
     {
-        /* Send the HTTP headers over the network. */
-        if( pRequestHeaders->headersLen > INT32_MAX )
-        {
-            LogError( ( "Parameter check failed: pRequestHeaders->headersLen > "
-                        "2147483647 (INT32_MAX)." ) );
-            returnStatus = HTTPInvalidParameter;
-        }
-        else
-        {
-            LogDebug( ( "Sending HTTP request headers: HeaderBytes=%lu",
-                        ( unsigned long ) ( pRequestHeaders->headersLen ) ) );
-            returnStatus = sendHttpData( pTransport, pRequestHeaders->pBuffer, pRequestHeaders->headersLen );
-        }
+        LogDebug( ( "Sending HTTP request headers: HeaderBytes=%lu",
```

Comment:
Why was the check on the previous line 1727 removed?  Is that an impossible condition here?

Reply:
This check was removed because the condition was there to satisfy some incorrect logic in the CBMC stub. 
We should need to limit the headers we are sending to less than size_t for the network interface send function.